packages/backend-for-frontend/src/services/agreementService.ts

```diff
@@ -760,20 +761,35 @@ async function getConsumerProducerEservice(
     params: { id: agreement.producerId },
     headers,
   });
+
   const eserviceTask = catalogProcessClient.getEServiceById({
     params: { eServiceId: agreement.eserviceId },
     headers,
   });
-  const [consumer, producer, eservice] = await Promise.all([
+
+  const delegationTask = delegationProcessClient.delegation.getDelegations({
+    queries: {
+      delegateIds: [agreement.consumerId],
```

Comment:
The consumer is the delegator, while the delegate should be the caller
```suggestion
      delegatorIds: [agreement.consumerId],
      delegateIds: [authData.organizationId]
```

Reply:
nice catch 👍